File: app/src/main/java/com/x7/steambadger/database/AbstractOpenHelper.java
package com.x7.steambadger.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper;
import com.j256.ormlite.support.ConnectionSource;

public abstract class AbstractOpenHelper extends OrmLiteSqliteOpenHelper {

	public AbstractOpenHelper(Context context, String DATABASE_NAME,
			int DATABASE_VERSION) {
		super(context, DATABASE_NAME, null, DATABASE_VERSION);
	}

	@Override
	public void onCreate(SQLiteDatabase db, ConnectionSource connectionSource) {
		createTables(db, connectionSource);
	}

	@Override
	public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource,
			int oldVersion, int newVersion) {
		dropTables(db, connectionSource);
		createTables(db, connectionSource);
	}

	public abstract void createTables(SQLiteDatabase db,
			ConnectionSource connectionSource);

	public abstract void dropTables(SQLiteDatabase db,
			ConnectionSource connectionSource);

	public abstract void clearTables(SQLiteDatabase db,
			ConnectionSource connectionSource);
}

File: app/src/main/java/com/dougmaitelli/steambadger/database/AbstractOpenHelper.kt
package com.dougmaitelli.steambadger.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase

import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper
import com.j256.ormlite.support.ConnectionSource

abstract class AbstractOpenHelper(context: Context, DATABASE_NAME: String,
                                  DATABASE_VERSION: Int) : OrmLiteSqliteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    override fun onCreate(db: SQLiteDatabase, connectionSource: ConnectionSource) {
        createTables(db, connectionSource)
    }

    override fun onUpgrade(db: SQLiteDatabase, connectionSource: ConnectionSource,
                  oldVersion: Int, newVersion: Int) {
        dropTables(db, connectionSource)
        createTables(db, connectionSource)
    }

    abstract fun createTables(db: SQLiteDatabase,
                              connectionSource: ConnectionSource)

    abstract fun dropTables(db: SQLiteDatabase,
                            connectionSource: ConnectionSource)

    abstract fun clearTables(db: SQLiteDatabase,
                             connectionSource: ConnectionSource)
}